app/javascript/components/supplemental_application/SupplementalApplicationPage.js

```diff
@@ -60,6 +60,8 @@ class SupplementalApplicationPage extends React.Component {
 
     await updateApplicationAction(synchedApplication)
```

Comment:
check the response here, if it's false, don't do the redirect.

Reply:
@fddayan I think that if there's an error an alert is raised, I don't think that it actually responds "false" if the update is unsuccessful.

Reply:
it looks like it returns  the response from the service and raises a `window.alert`. After you dismiss the alert it will redirect. I feel it would be good to NOT redirect if there is an error. 

form the code is doing 
```
  if (response === false) {
    Alerts.error()
  }
```

so if this code is correct, it's returning `false`

you can do 
```
if (response) {
 window.location.reload()
}
```
makes sense?